src/shared_utilities/interpolating_time_varying_input1d.jl

```diff
@@ -0,0 +1,165 @@
+import CUDA
+
+"""
+    NearestNeighbor
+
+Return the value corresponding to the point closest to the input time.
+"""
+struct NearestNeighbor <: AbstractInterpolationMethod end
+
+"""
+    LinearInterpolation
+
+Perform linear interpolation between the two neighboring points.
+"""
+struct LinearInterpolation <: AbstractInterpolationMethod end
+
+"""
+    InterpolatingTimeVaryingInput1D
+
+The constructor for InterpolatingTimeVaryingInput1D is not supposed to be used directly, unless you
+know what you are doing. The constructor does not perform any check and does not take care of
+GPU compatibility. It is reasonability of Integrator() to do so.
```

Comment:
could you clarify this? " It is reasonability of Integrator() to do so" What is Integrator()?

Reply:
Right, `Integrator` has nothing to do with this, it was a typo.

Reply:
Fixed.